osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcess.java

```diff
@@ -76,24 +84,35 @@ public void run() {
         LOGGER.info("Attempting key recovery for device {}", this.deviceIdentification);
 
         try {
-            this.initDevice();
+            this.findDevice();
         } catch (final Exception e) {
-            LOGGER.error("Unexpected exception: {}", e);
+            LOGGER.error("Could not find device", e);
+            //TODO why try to find device if you don't do anything with the result?!?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)

Reply:
Not good to leave TODOs in the code. Either figure out and resolve the issues, or if they are not pressing, manage them through the backlog. (Same for the other TODOs.)